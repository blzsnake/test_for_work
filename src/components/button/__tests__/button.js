import React from 'react';
import { shallow } from 'enzyme';

import Button from '../index';

const props = {
  tagName: 'button',
  type: 'button',
};

describe('Should render', () => {
  it('Button render', () => {
    const component = shallow(<Button {...props} />);

    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
