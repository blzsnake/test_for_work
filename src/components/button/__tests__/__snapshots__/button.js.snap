// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Button render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    disabled={false}
    menu={false}
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    tagName="button"
    type="button"
    wide={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "button",
      "disabled": false,
      "href": undefined,
      "onClick": [Function],
      "onMouseLeave": [Function],
      "onMouseOver": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "button",
        "disabled": false,
        "href": undefined,
        "onClick": [Function],
        "onMouseLeave": [Function],
        "onMouseOver": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
